### ISSUE TITLE
[CR121] Downloaded files (music, pics) can't be opened

### ISSUE THREAD

**Issue by hffvld** [brave/brave-browser#35363]
## Description 

Downloaded files (music, pics) can't be opened in Brave with `CR121`.

## Steps to reproduce 

   1. Install and launch Brave `1.63.120` with `CR121` or newer
   2. Open any website > Three-dot menu > `Share` button
   3. Tap `Long Screenshot` or `QR Code` > Proceed and Save/Download a file
   4. Three-dot menu > Downloads > Try to open a file > Observe

## Actual result 

Downloaded files (music, pics) can't be opened in Brave with `CR121`. The issue is not reproducible in Brave with `CR120`.

|1|2|3|
|:-:|:-:|:-:|
| ![1](https://github.com/brave/brave-browser/assets/128532965/2cd22b51-ddd4-454c-9e49-f34d469a5875) | ![2](https://github.com/brave/brave-browser/assets/128532965/9a108c54-5bc4-4c5a-b62b-0f8d3b11094b) | ![3](https://github.com/brave/brave-browser/assets/128532965/2bcc2367-03ee-4419-bd81-1463450b96b0) |

https://github.com/brave/brave-browser/assets/128532965/fd8d4a92-2aa7-461a-901e-75f3ed66f2f2

## Expected result

Downloaded files (music, pics) can be opened/launched.

https://github.com/brave/brave-browser/assets/128532965/cb02a3d4-8a4e-4a5b-b68b-fd47eebd7a82

## Issue reproduces how often 

Easily reproduced

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? NO
- Can you reproduce this issue with the current Play Store Beta version? NO
- Can you reproduce this issue with the current Play Store Nightly version? NO

## Device details
   - Install type (ARM, x86): ARM
   - Device type (Phone, Tablet, Phablet): Phone
   - Android version: Android 13 / 14
   
## Brave version

```
Brave build: 1.63.122
Chromium: 121.0.6167.57 (Official Build) canary (64-bit) 
```

### Website problems only
- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards?
- Is the issue reproducible on the latest version of Chrome?

**Comment by deeppandya** [comment]
please @kjozwiak @samartnik if it's fixed with 121.0.6167.75

**Comment by bsclifton** [comment]
Tracked upstream with http://crbug.com/1515491

Asked in Chromium Slack - there is an assignee and a fix is being worked on. Accidental regression from https://chromium-review.googlesource.com/c/chromium/src/+/5059490/6/chrome/android/java/src/org/chromium/chrome/browser/customtabs/content/CustomTabActivityTabController.java

**Comment by kjozwiak** [comment]
> Tracked upstream with http://crbug.com/1515491 (restricted to Google employees at the moment)
> 
> Asked in Chromium Slack - there is an assignee and a fix is being worked on. Accidental regression from https://chromium-review.googlesource.com/c/chromium/src/+/5059490/6/chrome/android/java/src/org/chromium/chrome/browser/customtabs/content/CustomTabActivityTabController.java

awesome, thanks for digging in @bsclifton 👍

**Comment by kjozwiak** [comment]
Closing/Moving the above into `1.62.x` as the above has been fixed with `121.0.6167.85` as per @samartnik. The above will require `1.62.150 Chromium: 121.0.6167.85` or higher for `1.62.x` verification 👍

**Comment by hffvld** [comment]
Verified on `Pixel 7` using version(s):
```
Device/OS: Pixel 7 / panther_beta-user 14 AP11.231215.007 release-keys
Brave build: 1.62.150
Chromium: 121.0.6167.85 (Official Build) (64-bit) 
```
---------------------------

**STEPS:**
1. Followed the steps from https://github.com/brave/brave-browser/issues/35363#issue-2085128027

**ACTUAL RESULTS:**
- Verified that downloaded files (audio and pics) can be opened
- Verified the same with a new and upgraded profile

---------------------------

|1|2|3|
|:-:|:-:|:-:|
| ![1](https://github.com/brave/brave-browser/assets/128532965/94489524-5d29-4a98-99e6-fdd07c89c279) | ![2](https://github.com/brave/brave-browser/assets/128532965/9fc19ccc-65ed-4a33-ae6d-a5c4ee475c7f) | ![3](https://github.com/brave/brave-browser/assets/128532965/f481b1cc-e568-44a6-aea9-4300afe6473a) |

*New profile*

https://github.com/brave/brave-browser/assets/128532965/6e00601c-1999-43da-86a0-556798276a66

*Upgraded profile*

https://github.com/brave/brave-browser/assets/128532965/b7935ae9-c832-49ba-9f21-c28fb2376f0c
